test/SlowTests/Issues/RavenDB-15539.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using FastTests;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_15539 : RavenTestBase
+    {
+        public RavenDB_15539(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private class User
+        {
+            public string Name;
+
+            [JsonIgnore]
+            public bool IgnoreChanges;
+        }
+
+        [Fact]
+        public void CanIgnoreChanges()
+        {
+            using var store = GetDocumentStore(new Options
+            {
+                ModifyDocumentStore = s =>
+                {
+                    s.Conventions.ShouldIgnoreEntityChanges
+                        = (sender, args) => args.ShouldIgnoreChanges =args.Entity is User u && u.IgnoreChanges;
```

Comment:
For me it would be much more natural to just return `bool` here instead of assigning it to `ShouldIgnoreChanges`. The convention name is `ShouldIgnoreEntityChanges` anyway so the following is simpler:
```
 s.Conventions.ShouldIgnoreEntityChanges
                        => (id, entity) => entity is User u && u.IgnoreChanges;

```